jenkins-master/jobs/scripts/workspace/runtests.py

```diff
@@ -5,85 +5,84 @@
+                              help='The expected version of the updated build')
+    return parser.parse_args()
 
-    def run_tests(self, args):
-        settings = config['test_types'][args.type]
 
-        # In case the log folder does not exist yet we have to create it because
-        # otherwise Marionette will fail for e.g. gecko.log (see bug 1211666)
-        if settings['logs'].get('gecko.log'):
-            try:
-                os.makedirs(os.path.dirname(settings['logs']['gecko.log']))
-            except OSError:
-                print('Failed to create log folder for {}'.format(settings['logs']['gecko.log']))
+def main():
+    args = parse_args()
 
-        print('Downloading the installer: {}'.format(args.installer_url))
-        scraper = FactoryScraper('direct',
-                                 url=args.installer_url,
-                                 retry_attempts=5,
-                                 retry_delay=30,
-                                 )
-        installer_path = scraper.download()
+    try:
+        settings = config['test_types'][args.test_type]
+        retval = 0
 
-        command = [
-            'firefox-ui-update' if args.type == 'update' else 'firefox-ui-tests',
-            '--installer', installer_path,
-            '--workspace', os.getcwd(),
-            '--log-tbpl', settings['logs']['tbpl.log'],
-        ]
+        command = [os.path.join('mozharness', 'scripts', settings['harness_script']),
+                   '--cfg', settings['harness_config'],
+                   '--firefox-ui-branch', args.repository,
+                   '--installer-url', args.installer_url,
+                   ]
 
-        if args.type == 'update':
-            # Enable Gecko log to the console because the file would be overwritten
-            # by the second update test
-            command.extend(['--gecko-log', '-'])
+        if args.test_type == 'functional':
+            # No additional arguments needed
+            pass
 
+        elif args.test_type == 'update':
             if args.update_channel:
                 command.extend(['--update-channel', args.update_channel])
             if args.update_target_version:
                 command.extend(['--update-target-version', args.update_target_version])
             if args.update_target_build_id:
-                command.extend(['--update-target-buildid', args.update_target_build_id])
+                command.extend(['--update-target-build-id', args.update_target_build_id])
 
-        elif args.type == 'functional':
-            command.extend(['--gecko-log', settings['logs']['gecko.log']])
-
-            manifests = [firefox_puppeteer.manifest, firefox_ui_tests.manifest_functional]
-            command.extend(manifests)
-
-        retval = 0
+        env = copy.copy(os.environ)
+        for var in ENV_VARS_TO_PURGE:
+            env.pop(var, None)
 
         print('Calling command to execute tests: {}'.format(command))
```

Comment:
>>          print('Calling command to execute tests: {}'.format(command))
> 
> print? Shouldn't this be some kind of logger?

We can do but it won't end-up in a log file we upload. If you think it
is important now I can do it. Otherwise we can file a follow-up issue
maybe mentored. Please let me know.